### PR TITLE
Add a prev and next link to lessons template

### LINKED ISSUE #5: Add Previous and Next lesson links.
The lesson is already set up to contain the next and previous lessons, we just need to add the UI. Additionally I parsed all the lessons and pulled the first paragraph (above the TOC) out and into the `excerpt` attribute. This allows us to display this along with these links.

I'll work on tackling this next.

### PR DESCRIPTION
**Pull request by burden**
This one closes #5 by adding a section to the footer for the next and previous lesson. 

es-prevnext-mobile
es-prevnext-full

### CODE REVIEW

File: assets/css/app.css [hunk 1]
```diff
@@ -2,6 +2,15 @@
 @import "tailwindcss/components";
 @import "tailwindcss/utilities";
 
+.full-width {
```
Comment (doomspork) [inline assets/css/app.css:5]: Is it possible to achieve this with only tailwind classes? Just thinking about long term

**Review by doomspork — COMMENTED** [review]
This is awesome, thank you @burden! One question

File: assets/css/app.css (hunk 1 above)
Comment (burden) [reply assets/css/app.css:5]: Challenge accepted :muscle:

File: lib/school_house_web/templates/lesson/lesson.html.eex [hunk 2]
```diff
@@ -13,10 +13,67 @@
...
+              </p>
+              <p class="text-italic md:text-left text-center text-sm text-gray-500 overflow-hidden max-h-16">
+                <%= raw @lesson.next.excerpt %>
+              </p>
+            </div>
+          <% end %>
+        </div>
+      </div>
+  </div>
+</section>
```
Comment (doomspork) [inline lib/school_house_web/templates/lesson/lesson.html.eex]: If _you_ wanted we could pull this out into a partial `lib/school_house_web/templates/lesson/_pagination.html.eex`, then update this to:

```eex
<%= render("_pagination.html", previous: @lesson.previous, next: @lesson.next) %>
```

You would need to update the markup so `@lesson.next` became `@next` etc.

File: assets/tailwind.config.js [hunk 3]
```diff
@@ -18,6 +18,9 @@ module.exports = {
             container: {
                 center: true,
             },
+            margin: {
+            'half-screen': '-50vw',
+            },
```
Comment (doomspork) [inline assets/tailwind.config.js:23]: I smell a short blog post :wink:

File: lib/school_house_web/templates/lesson/lesson.html.eex [hunk 4]
```diff
@@ -13,10 +13,67 @@
...
```
Comment (doomspork) [inline lib/school_house_web/templates/lesson/lesson.html.eex:73]: If you wanted we could pull this out into a partial like `lib/school_house_web/templates/lesson/_pagination.html.eex`. Then we could update this to:

```eex
<%= render("_pagination.html", previous: @lesson.previous, next: @lesson.next) %>
```

You'll need to update the markup in the partial from things like `@lesson.next` to `@next`.

File: lib/school_house_web/templates/lesson/lesson.html.eex [hunk 5]
```diff
@@ -13,10 +13,67 @@
...
+                <%= raw @lesson.previous.excerpt %>
+              </p>
+            </div>
+          <% end %>
+        </div>
+        <div class="md:w-1/3">
+          <%= if @lesson.next do %>
+            <div class="flex md:justify-end justify-center items-center content-center mb-6">
+              <h3 class="leading-tight text-sm font-semibold text-gray-400 tracking-wider uppercase">
+                <a href="/<%= @lesson.next.locale %>/<%= @lesson.next.section %>/<%= @lesson.next.name %>" class="hover:text-gray-900" rel="next">
```
Comment (doomspork) [inline lib/school_house_web/templates/lesson/lesson.html.eex:53]: Let's use a link helper for this. I think `lesson_link/4` in `html_helpers.ex` might do the trick:


```eex
<%= lesson_link(@conn, @lesson.next.section, @lesson.next.name, "hover:text-gray-900"), do: @lesson.next.title %>
```

**Review by doomspork — COMMENTED** [review]
I'm not sure what happened. GitHub borked and lost a few of my comment :sob:

File: lib/school_house_web/templates/lesson/lesson.html.eex (hunk 5 above)
Comment (doomspork) [reply lib/school_house_web/templates/lesson/lesson.html.eex:53]: We also have some "brand" colors defined in `tailwind.config.js` that could potentially be better here than `text-gray-900` but that's not a big one.

File: assets/tailwind.config.js (hunk 3 above)
Comment (burden) [reply assets/tailwind.config.js:23]: this and friends allow us to break out of the width defined on the `<main>` parent. File this one under "hacks"

**Review by doomspork — APPROVED** [review]

**Pull request merged**